### ISSUE TITLE
Don't check string with newline

### ISSUE THREAD

**Issue by PbIXTOP** [sissaschool/xmlschema#223]
I have xml string element with new line
```xml
        <CadastralBlock>38:36:000031
</CadastralBlock>
```
In XSD CadastralBlock define as 
```xml
<xs:element name="CadastralBlock" type="CadastralBlockType">
</xs:element>
<xs:simpleType name="CadastralBlockType">
   <xs:restriction base="xs:string">
        <xs:maxLength value="13"/>
        <xs:minLength value="12"/>
        <xs:pattern value="\d{2}:\d{2}:\d{6,7}"/>
    </xs:restriction>
</xs:simpleType>
```
If i use xmlschema for validate it not show  error
if use xmllint or etree.XMLscheme for validate get error
`element CadastralBlock: Schemas validity error : Element 'CadastralBlock': [facet 'pattern'] The value '38:36:000031
' is not accepted by the pattern '\d{2}:\d{2}:\d{6,7}'`

version xmlschema==1.3.1

**Comment by brunato** [comment]
Hi,
it's a problem with different behaviour of the regex anchor '$' in XML Schema regular expressions, that cannot match strings ending with \n, so the pattern has to be translated to the Python's regex pattern `^(?:\d{2}:\d{2}:\d{6,7})$(?!\n\Z)` instead of `^(?:\d{2}:\d{2}:\d{6,7})$`.

The regex processor is located in elementpath package, so the fix will be included in the next release that package.

**Comment by brunato** [comment]
Fixed with _elementpath_ v2.1.2:

```
$ pip list
Package     Version
----------- -------
elementpath 2.1.2
lxml        4.6.2
pip         20.3.3
setuptools  51.0.0
wheel       0.36.2
xmlschema   1.3.1
```

```pycon
>>> import xmlschema
>>> xs = xmlschema.XMLSchema("issue_223.xsd")
>>> xs.validate("issue_223.xml")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/brunato/.virtualenvs/xmlschema-v131/lib/python3.9/site-packages/xmlschema/validators/schema.py", line 1418, in validate
    raise error
xmlschema.validators.exceptions.XMLSchemaValidationError: failed validating '38:36:000031\n' with XsdPatternFacets(['\\d{2}:\\d{2}:\\d{6,7}']):

Reason: value doesn't match any pattern of ['\\d{2}:\\d{2}:\\d{6,7}'].

Schema:

  <xs:pattern xmlns:xs="http://www.w3.org/2001/XMLSchema" value="\d{2}:\d{2}:\d{6,7}" />

Instance:

  <CadastralBlock>38:36:000031
  </CadastralBlock>

Path: /CadastralBlock
```
